### DISCUSSION TITLE
ESP Logging and TAG Macro definition

### DISCUSSION THREAD

**Discussion by APF-MRoger** [pschatzmann/arduino-audio-tools#1804] [Q&A] [answered]
### Problem Description

This is more a question than a bug report. I noticed that in some parts of this lib and in the audio-driver one as well, you used `const char* TAG` instead of a `#define TAG "someTag"` macro (with macro shielding etc). That's causing a _mild_ pain when needing to interact with the original ESP_IDF_LOG (in my case because I am redirecting logs to an SD card instead of UART0). Is there a reason to use the char* instead of the macro ?

### Device Description

ESP32 AudioKit v2.2, if relevant.

### Sketch

```cpp
;Only an issue when using ESP_IDF_LOG in platformio.ini with following build_flags
-D CORE_DEBUG_LEVEL=4
-D USE_ESP_IDF_LOG
-DTAG="\"ARDUINO\""
```

### What is your development environment

PlatformIo + Arduino Framework + espressif32@6.9.0 + -D USE_ESP_IDF_LOG in platformIo.

### I have checked existing issues, discussions and online documentation

- [X] I confirm I have checked existing issues, discussions and online documentation

**Comment by pschatzmann** [comment, marked as answer]
In the AudioTools this was not used, so I removed it.
I did not find this anywhere in the audio-driver though...

**Reply by APF-MRoger** [reply]
Let me try again then, I might have missed something ...

**Reply by APF-MRoger** [reply]
So... I must have confused myself or accidentally used an older version of the libraries. I just imported my project to a different computer, built it fresh from the ground up in Platformio, removed the `#ifdef TAG #undef TAG #endif` I had sawn in a bit everywhere, and it compiles fine. I'll get back to the original computer this failed on, tonight, and will try to recompile.

For what it's worth with Audio tools on 0.9.9+sha5d41c41 and audio-driver at 0.1.0+sha0fe3454 it seems to work just fine.
Apologies for the confusion !

**Reply by martinroger** [reply]
Confirmed (from the other account, other computer). Did a full clean, reloaded the latest b6c3025 and it works all ok. May this help another forgetful one like me : remember to clean your libdeps from time to time.
